test/unit/nics/bond_test.rb

```diff
@@ -62,4 +62,16 @@ class BondTest < ActiveSupport::TestCase
     assert bond.remove_slave('eth2')
     assert_equal [], bond.attached_devices_identifiers
   end
+
+  test 'identifier is required for managed bonds' do
+    bond = FactoryGirl.build(:nic_bond, :attached_devices => 'eth0,eth1,eth2', :managed => true, :identifier => '')
+    refute bond.valid?
```

Comment:
These `bond.valid?` would still be false if you remove the validation, as there's another validation for `host`. You need to assign these a host.